openmrs-client/src/main/java/org/openmrs/mobile/activities/addeditpatient/SimilarPatientsRecyclerViewAdapter.java

```diff
@@ -70,19 +74,25 @@ public void onBindViewHolder(PatientViewHolder holder, int position) {
         holder.mRowLayout.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                if (!(new PatientDAO().isUserAlreadySaved(patient.getUuid()))) {
-                    downloadPatient(patient);
-                }
-                Intent intent = new Intent(mContext, PatientDashboardActivity.class);
-                intent.putExtra(ApplicationConstants.BundleKeys.PATIENT_ID_BUNDLE, getPatientId(patient));
-                mContext.startActivity(intent);
-                mContext.finish();
+
+                patientDao.isUserAlreadySaved(patient.getUuid())
+                        .observeOn(AndroidSchedulers.mainThread())
+                        .subscribe(isSaved -> {
+                            if (!isSaved) {
+                                downloadPatient(patient);
+                            }
+                            Intent intent = new Intent(mContext, PatientDashboardActivity.class);
```

Comment:
This can be outside of the subscribe block I guess ? Plus you can use filter instead of if check